engine/classes/Elgg/UpgradeService.php

```diff
@@ -14,155 +14,168 @@
  */
 class UpgradeService {
 
-	/**
-	 * Global Elgg configuration
-	 *
-	 * @var \stdClass
-	 */
-	private $CONFIG;
+	private $events;
+
+	private $translator;
+
+	private $datalist;
+
+	private $logger;
+
+	private $db;
+
+	private $upgrade;
+
+	private $batch_run_time_in_secs;
 
 	/**
 	 * Constructor
 	 */
 	public function __construct() {
-		global $CONFIG;
-		$this->CONFIG = $CONFIG;
+		$this->events = _elgg_services()->events;
+		$this->translator = _elgg_services()->translator;
+		$this->datalist = _elgg_services()->datalist;
+		$this->logger = _elgg_services()->logger;
+		$this->db = _elgg_services()->db;
+		$this->session = _elgg_services()->session;
```

Comment:
I think I'm going to add a separate service just for locating the upgrade files. That will hopefully reduce the amount of needed services and make it easier to create unit tests.